File: app/src/main/java/com/rakangsoftware/architecture/data/ContactRepository.java
package com.rakangsoftware.architecture.data;

import android.arch.lifecycle.LiveData;

import java.util.List;

public class ContactRepository {

    private final ContactDao mContactDao;

    public ContactRepository(final ContactDao contactDao) {
        mContactDao = contactDao;
    }

    public LiveData<List<Contact>> getAll() {
        return mContactDao.getAllContacts();
    }

    public void create(final Contact contact) {
        mContactDao.insert(contact);
    }

    public void delete(final Contact contact) {
        mContactDao.delete(contact);
    }
}

File: app/src/main/java/com/rakangsoftware/architecture/data/ContactRepository.kt
package com.rakangsoftware.architecture.data

import android.arch.lifecycle.LiveData

class ContactRepository(private val mContactDao: ContactDao) {

    val all: LiveData<List<Contact>>
        get() = mContactDao.allContacts

    fun create(contact: Contact) {
        mContactDao.insert(contact)
    }

    fun delete(contact: Contact) {
        mContactDao.delete(contact)
    }
}
